app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationLoadStateAdapter.kt

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2021 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.conversation
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.LoadState
+import androidx.paging.LoadStateAdapter
+import com.keylesspalace.tusky.adapter.NetworkStateViewHolder
+import com.keylesspalace.tusky.databinding.ItemNetworkStateBinding
+
+class ConversationLoadStateAdapter(
```

Comment:
I would appreciate few words on what it's for

Reply:
Its an adapter that shows the loading state (nothing, spinner, or retry button), in this case at the bottom of the loaded items. It works by utilizing ConcatAdapter and a load state listener. Pretty nice actually, previously I had to code that myself.

Reply:
Thanks, I meant it should probably be a comment in code but see for yourself ^^